cou/apps/base.py

```diff
@@ -538,24 +554,32 @@ def _get_charmhub_migration_step(self) -> PreUpgradeStep:
             ),
         )
 
-    def _get_change_to_openstack_channels_step(self) -> PreUpgradeStep:
+    def _get_change_to_openstack_channels_step(self, target: OpenStackRelease) -> PreUpgradeStep:
         """Get the step for changing to OpenStack channels.
 
+        :param target: OpenStack release as target to upgrade.
+        :type target: OpenStackRelease
         :return: Step for changing to OpenStack channels
         :rtype: PreUpgradeStep
         """
+        expected_current_channel = (
```

Comment:
Should this if condition be part of the property? Or can we rename this variable?
Doing `expected_current_channel = (self.expected_current_channel if ... else ...`  do not look, right, what is expected channel then?